ENH: Add alternate threshold effect

Sunderlandkyl opened the pull request:
New threshold effect is a subclass of the threshold effect in the Slicer core that implements a new flood filling type of threshold.
Once the user has selected a threshold range + optional roi, they can ctrl+click on the image to add the region under the mouse to the image.

The voxels added to the segment are calculated using one of three methods:
* Masking
* GrowCut
* WaterShed

lassoan commented on SegmentEditorThreshold2/SegmentEditorThreshold2Lib/SegmentEditorEffect.py:
```diff
@@ -0,0 +1,406 @@
+<ul style="feature: 0">
+<li><b>Use for masking:</b> set the selected intensity range as <dfn>Editable intensity range</dfn> and switch to Paint effect.</li>
+<li><b>Apply:</b> set the previewed segmentation in the selected segment. Previous contents of the segment is overwritten.</li>
+</ul><p></html>"""
```
Add some info here about how to use it (control+click, ...)

lassoan commented on CMakeLists.txt:
```diff
@@ -30,6 +30,7 @@ add_subdirectory(SegmentEditorSurfaceCut)
 add_subdirectory(SegmentEditorWatershed)
 add_subdirectory(SegmentEditorDrawTube)
 add_subdirectory(SegmentEditorSplitVolume)
+add_subdirectory(SegmentEditorThreshold2)
```
"LocalThreshold" may be a better name.

Sunderlandkyl committed:
ENH: Add alternate threshold effect

New threshold effect is a subclass of the threshold effect in the Slicer core that implements a new flood filling type of threshold.
Once the user has selected a threshold range + optional roi, they can ctrl+click on the image to add the region under the mouse to the image.

The voxels added to the segment are calculated using one of three methods:
* Masking
* GrowCut
* WaterShed

Sunderlandkyl commented:
- Renamed to Local Threshold.
- Reduced opacity of preview
- Updated help text
- New icon

lassoan commented:
Does this work with Slicer-4.10?

Sunderlandkyl commented:
Yes

lassoan commented:
Perfect, thank you. Merged.
